### PR TITLE
Switch CI to nf-core/setup-nextflow action

### PR DESCRIPTION
**Pull request by MillironX**
Switches the workflows in this repo and the template to point to use [setup-nextflow](https://github.com/MillironX/setup-nextflow), rather than try to manually install Nextflow in a `run:` block. This allows for simplifying the execution matrix, and removes the need to keep track of whether a stable or edge release is actually the most recent.

## PR checklist

- [x] This comment contains a description of changes (with reason)
- [x] `CHANGELOG.md` is updated
- [na] If you've fixed a bug or added code that should be tested, add tests!
- [na] Documentation in `docs` is updated

### CODE REVIEW

**Review by ewels — APPROVED** [review]
Nice!

Can maybe also remove the `CAPSULE_LOG: none` env vars too (see https://github.com/MillironX/setup-nextflow/issues/1 )

Also update if we're moving to nf-core, as discussed in Slack 👍🏻 

Otherwise LGTM! 💥

**Comment by MillironX** [comment]
Converting to draft until two things are fixed:

1. Linting checks need to target the new matrix format
2. Action should point to https://github.com/nf-core/setup-nextflow

File: nf_core/pipeline-template/.github/workflows/ci.yml [hunk 1]
```diff
@@ -30,7 +30,7 @@ jobs:
       - name: Install Nextflow
         uses: nf-core/setup-nextflow@v1
         with:
-          version: ${{ matrix.NXF_VER }}
+          version: "{% raw %}${{{% endraw %} matrix.NXF_VER {% raw %}}}{% endraw %}"
```
Comment (ewels) [inline nf_core/pipeline-template/.github/workflows/ci.yml:33]:
```suggestion
          version: "{% raw %}${{ matrix.NXF_VER }}{% endraw %}"
```

File: tests/test_bump_version.py [hunk 2]
```diff
@@ -75,7 +75,7 @@ def test_bump_nextflow_version(datafiles, tmp_path):
     # Check .github/workflows/ci.yml
     with open(new_pipeline_obj._fp(".github/workflows/ci.yml")) as fh:
         ci_yaml = yaml.safe_load(fh)
-    assert ci_yaml["jobs"]["test"]["strategy"]["matrix"]["include"][0]["NXF_VER"] == version
+    assert ci_yaml["jobs"]["test"]["strategy"]["matrix"]["NXF_VER"][0] == version
```
Comment (ewels) [inline tests/test_bump_version.py:78]: I feel like the actual code for `nf-core bump versions` should also need updating, no? Or do the existing regexes still work somehow?

**Review by ewels — APPROVED** [review]
Just a few lurking `CAPSULE_LOG` env vars left I think? Otherwise the code LGTM 👍🏻

File: tests/test_bump_version.py (hunk 2 above)
Comment (MillironX) [reply tests/test_bump_version.py:79]: :facepalm: Yes it does. See d52a173

Comment (ewels) [reply tests/test_bump_version.py:79]: Correct 👍🏻 The installation log is very noisy, just 100s of lines of download progress which is not informative in a non-interactive log.

Aim is to keep the CI workflows as clean as possible, so would be good to move it into the action and have it set by default.

Comment (MillironX) [reply tests/test_bump_version.py:79]: Done. See 0bab55b

Comment (ewels) [reply tests/test_bump_version.py:79]: Great! Still needs implementing upstream though, right? https://github.com/nf-core/setup-nextflow/issues/1

**Pull request merged**